src/components/reservation/helper.ts

```diff
@@ -17,28 +17,35 @@ import {
 import { Manifestation } from "../../core/utils/types/entities";
 import { PeriodicalEdition } from "../material/periodical/helper";
 
-export const smsNotificationsIsEnabled = (
-  configValue: string | undefined | string[]
-) => configValue === "1";
+export const strToBool = (configValue: string | undefined | string[]) =>
+  configValue === "1";
 
 export const getPreferredBranch = (id: string, array: AgencyBranch[]) => {
   const locationItem = array.find((item) => item.branchId === id);
   return locationItem ? locationItem.title : id;
 };
+export const getInterestPeriods = (t: UseTextFunction) => {
+  const visibleInterestPeriods: { [key: string]: string } = {};
+  const interestPeriods = [
+    ["interestPeriodOneMonthConfigText", "30", "oneMonthText"],
+    ["interestPeriodTwoMonthsConfigText", "60", "twoMonthsText"],
+    ["interestPeriodThreeMonthsConfigText", "90", "threeMonthsText"],
+    ["interestPeriodSixMonthsConfigText", "180", "sixMonthsText"],
+    ["interestPeriodOneYearConfigText", "360", "oneYearText"]
+  ];
 
-export const hardcodedInterestPeriods = (t: UseTextFunction) => {
-  return {
-    "30": t("oneMonthText"),
-    "60": t("twoMonthsText"),
-    "90": t("threeMonthsText"),
-    "180": t("sixMonthsText"),
-    "360": t("oneYearText")
-  };
+  interestPeriods.forEach(([config, key, text]) => {
+    if (strToBool(t(config))) {
```

Comment:
This is really cryptic to me :) Why are you translating some config and then checking if it is `"1"`?
Could be a valid reason. But I get the impression that it could be done otherwise.

Reply:
The config is a bool, configurable for each interest period. The bool is set up as an "argtype", will be defined as a checkbox in drupal, either returning "1" or "0", ultimately making the visibility of each interest period configuarable in drupal